bot/cogs/codeblock/parsing.py

```diff
@@ -0,0 +1,224 @@
+"""This module provides functions for parsing Markdown code blocks."""
+
+import ast
+import logging
+import re
+import textwrap
+from typing import NamedTuple, Optional, Sequence
+
+from bot import constants
+from bot.utils import has_lines
+
+log = logging.getLogger(__name__)
+
+BACKTICK = "`"
+PY_LANG_CODES = ("python", "py")  # Order is important; "py" is second cause it's a subset.
```

Comment:
We should add `pycon` here I think, python console is what it is for.

<img width="191" alt="Screenshot 2020-09-20 at 14 22 56" src="https://user-images.githubusercontent.com/20439493/93712281-cc2fea80-fb4c-11ea-8312-6b66d993e25e.png">

```suggestion
PY_LANG_CODES = ("python", "pycon", "py")  # Order is important; "py" is second cause it's a subset.
```

Reply:
I did not even know that this is a thing, but it sounds like a reasonable addition.

Reply:
Added in ace40ec. I wanted to also include a separate example with the `>>>` prefix so that the syntax highlighting works. However, I am going to hold off due to a bug in `escape_markdown` that doesn't escape `>` characters after newlines. I don't feel like escaping it myself.